server/src/main/java/org/elasticsearch/index/store/Store.java

```diff
@@ -1495,7 +1504,16 @@ public void trimUnsafeCommits(final Path translogPath) throws IOException {
 
                     // The new commit will use segment files from the starting commit but userData from the last commit by default.
                     // Thus, we need to manually set the userData from the starting commit to the new commit.
-                    writer.setLiveCommitData(startingIndexCommit.getUserData().entrySet());
+                    final Map<String, String> userData = startingIndexCommit.getUserData();
+                    writer.setLiveCommitData(() -> {
+                        if (userData.containsKey(ES_VERSION)) {
```

Comment:
I think this is questionable. If we trim the commit, we store it by the new ES version, which could be on a newer lucene version as well. With this, I think we risk seeing an ES version for an older release but in a commit that is a new lucene version. I think we should avoid that and simply always put in the current ES version. I also think trying to optimize this by not rebuilding the hashmap here is not worth it, this is not performance sensitive code.